libvirt/domain_metadata.go

```diff
@@ -0,0 +1,73 @@
+package libvirt
+
+import (
+	"encoding/xml"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/libvirt/libvirt-go-xml"
+)
+
+// TerraformInstanceXML type
+type TerraformInstanceXML struct {
+	XMLName xml.Name          `xml:"https://terraform.io instance"`
+	Tags    []TerraformTagXML `xml:"tag"`
+}
+
+// TerraformTagXML type
+type TerraformTagXML struct {
+	Key   string `xml:"name,attr"`
+	Value string `xml:",chardata"`
+}
+
+func tagsToXML(tags map[string]interface{}, metadata *TerraformInstanceXML) (out string, err error) {
+	// Overwrite existing tags while keeping additional metadata
+	metadata.Tags = []TerraformTagXML{}
+	for key, value := range tags {
+		metadata.Tags = append(metadata.Tags, TerraformTagXML{
+			Key:   key,
+			Value: value.(string),
+		})
+	}
+	var bytesOut []byte
+	if bytesOut, err = xml.Marshal(metadata); err != nil {
+		return "", fmt.Errorf("Failed to marshal metadata XML: %s", err)
+	}
+	return string(bytesOut), nil
+}
+
+func setMetadata(d *schema.ResourceData, domain *libvirtxml.Domain) (err error) {
```

Comment:
I think we should name this `setMetadataFromXML`, because only `setMetadata` is from an extnernal reviw kind the same of `domain.setMetadata` function.

Also `getMetadataFromXML` make more sense imho

Reply:
Don't really care about the precise function name. I renamed it.

Reply:
thx!